File: Math and Geometry/Python/Roman to Integer.py
# Problem link: https://leetcode.com/problems/roman-to-integer/description/
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_numerals = {"M": 1000, "CM": 900, "D": 500, "CD": 400, "C": 100, "XC": 90, "L": 50, "XL": 40,
                            "X": 10, "IX": 9, "V": 5, "IV": 4, "I": 1}
        
        result = idx = 0

        while idx < len(s):
            if idx + 1 < len(s) and s[idx: idx + 2] in roman_numerals:
                result += roman_numerals[s[idx: idx + 2]]
                idx += 2
            else:
                result += roman_numerals[s[idx]]
                idx += 1
        
        return result

File: Math and Geometry/Java/Roman to Integer.java
// problem link: https://leetcode.com/problems/roman-to-integer/description/
class Solution {
    public int romanToInt(String s) {
        Map<String, Integer> romanNumerals = new HashMap<>();
        romanNumerals.put("M", 1000);
        romanNumerals.put("CM", 900);
        romanNumerals.put("D", 500);
        romanNumerals.put("CD", 400);
        romanNumerals.put("C", 100);
        romanNumerals.put("XC", 90);
        romanNumerals.put("L", 50);
        romanNumerals.put("XL", 40);
        romanNumerals.put("X", 10);
        romanNumerals.put("IX", 9);
        romanNumerals.put("V", 5);
        romanNumerals.put("IV", 4);
        romanNumerals.put("I", 1);

        int result = 0, idx = 0;

        while(idx < s.length()) {
            if(idx + 1 < s.length() && romanNumerals.containsKey(s.substring(idx, idx + 2))) {
                result += romanNumerals.get(s.substring(idx, idx + 2));
                idx += 2;
            }
            else {
                result += romanNumerals.get(s.substring(idx, idx + 1));
                idx += 1;
            }
        }

        return result;
    }
}
